src/main/java/org/toradocu/extractor/BlockTag.java

```diff
@@ -11,7 +10,7 @@
  * specification (available after the translation of the comment). A specification is the
  * translation of the comment into a Java boolean condition.
  */
-public abstract class BlockTag<S extends Specification> {
+public abstract class BlockTag {
```

Comment:
So, remove the specifications from the tag classes, and have the translators create the specification from the tag.